php/phpcs/pre-commit

```diff
@@ -0,0 +1,117 @@
+#!/bin/bash
+# PHP CodeSniffer pre-commit hook for git
+#
+# @author Soenke Ruempler <soenke@ruempler.eu>
+# @author Sebastian Kaspari <s.kaspari@googlemail.com>
+#
+# see the README
+
+PHPCS_BIN=/usr/bin/phpcs
+PHPCS_CODING_STANDARD=PEAR
+PHPCS_IGNORE=
+TMP_STAGING=".tmp_staging"
+
+# parse config
+CONFIG_FILE=$(dirname $0)/config
+if [ -e $CONFIG_FILE ]; then
+    . $CONFIG_FILE
+fi
+
+# simple check if code sniffer is set up correctly
+if [ ! -x $PHPCS_BIN ]; then
+    echo "PHP CodeSniffer bin not found or executable -> $PHPCS_BIN"
+    exit 1
+fi
+
+# stolen from template file
+if git rev-parse --verify HEAD
+then
+    against=HEAD
+else
+    # Initial commit: diff against an empty tree object
+    against=4b825dc642cb6eb9a060e54bf8d69288fbee4904
```

Comment:
This is the empty tree sha1 given by git. More info [here](http://stackoverflow.com/questions/9765453/gits-semi-secret-empty-tree).